include/ros1_bridge/convert_builtin_interfaces.hpp

```diff
@@ -31,27 +31,26 @@ namespace ros1_bridge
 template<>
 void
 convert_1_to_2(
-  const ros::Duration & ros1_msg,
+  const ros::Duration & ros1_type,
```

Comment:
This change doesn't make sense to me. Can you clarify why you changed only one of the names of this prototype? and that the names don't match what's used in the implementation at https://github.com/ros2/ros1_bridge/pull/106/files#diff-e7604d29ef87753b67dcf843c9e940caR72?

Reply:
these are not messages but ROS 1 types hence the change in name. The prototype you're pointing to is taking messages and not types so using the `_msg` there is appropriate

Reply:
This patch changes the `ros1_msg` argument to `ros1_type` argument for all `convert_1_to_2` / `convert_2_to_1` functions in the header as well as the source file. The rational is that neither of the two arguments `ros::Duration` nor `ros::Time` is a message but just a plain type.

The function you are referencing is unrelated to this signature. That one is for actual message in the `builtin_interfaces` packages and therefore those arguments still use `_msg` as the argument suffix.

Reply:
ok, I'll keep looking at this to get my head around it myself, in the meantime the approval from @mikaelarguedas suggests it's just my misunderstanding so not a blocker :smile:

Reply:
following up, @mikaelarguedas pointed out what I was missing, which is that these particular msgs, fundamentally, are represented differently in ROS 1 and ROS 2: time/duration isn't a message in ROS 1, it is a data type, whereas in ROS 2 time/duration has message definitions associated with them. The contents of `ros1_msg.data` at [this line](https://github.com/ros2/ros1_bridge/pull/106/files#diff-e7604d29ef87753b67dcf843c9e940caR75) is not another message.